modules/services/repositories/src/main/java/au/com/shiftyjelly/pocketcasts/repositories/playback/PlaybackManager.kt

```diff
@@ -2226,11 +2228,34 @@ open class PlaybackManager @Inject constructor(
                 if (isPlaying()) {
                     statsManager.addTotalListeningTime(UPDATE_TIMER_POLL_TIME)
                 }
+                if (playbackStateRelay.blockingFirst().isSleepTimerRunning && !sleepAfterEpisode) { // Does not apply to end of episode sleep time
+                    setupFadeOutWhenFinishingSleepTimer()
+                }
             }
             .switchMapCompletable { updateCurrentPositionRx() }
             .subscribeBy(onError = { Timber.e(it) })
     }
 
+    private fun setupFadeOutWhenFinishingSleepTimer() {
+        // Needs to run in main thread because of player getVolume
+        Handler(Looper.getMainLooper()).post {
+            val timeLeft = sleepTimer.timeLeftInSecs()
+            timeLeft?.let {
+                val fadeDuration = 5
+                val maxVolume = (player as? SimplePlayer)?.getVolume() ?: 1.0f
+                val minVolume = 0.0f
+
+                if (timeLeft <= fadeDuration) {
+                    val fraction = timeLeft.toFloat() / fadeDuration
+                    val newVolume = minVolume + (maxVolume - minVolume) * fraction
```

Comment:
Isn't `minVolume` always `0` here?

Reply:
yeah. It's not necessary anymore. Let me remove it